cms/djangoapps/contentstore/views/course.py

```diff
@@ -849,16 +850,18 @@ def _create_or_rerun_course(request):
     Returns the destination course_key and overriding fields for the new course.
     Raises DuplicateCourseError and InvalidKeyError
     """
-    if not auth.user_has_role(request.user, CourseCreatorRole()):
-        raise PermissionDenied()
-
     try:
         org = request.json.get('org')
         course = request.json.get('number', request.json.get('course'))
         display_name = request.json.get('display_name')
         # force the start date for reruns and allow us to override start via the client
         start = request.json.get('start', CourseFields.start.default)
         run = request.json.get('run')
+        is_course_creator = (auth.user_has_role(request.user, CourseCreatorRole()) or
+                             auth.user_has_role(request.user, OrgContentCreatorRole(org=org)))
+
+        if not is_course_creator:
+            raise PermissionDenied()
```

Comment:
@farhaanbukhsh, this will show the `Network error` to the user in the Studio UI and that is not very clear. It will be better to have a clear error message when a user cannot create a course under an organization if he/she doesn't have the course creation privilege for that organization.